openedx/core/djangoapps/programs/utils.py

```diff
@@ -257,6 +257,12 @@ def reshape(course_run):
             Modify the structure of a course run dict to facilitate comparison
             with course run certificates.
             """
+            course_run_type = course_run['type']
+
+            # Treat no-id-professional enrollments as professional
+            if course_run_type == 'no-id-professional':
+                course_run_type = 'professional'
```

Comment:
Consider using the [constants](https://github.com/edx/edx-platform/blob/master/common/djangoapps/course_modes/models.py#L112-L117) on the `CourseMode` model.